src/gui/widgets/window.cpp

```diff
@@ -1307,12 +1308,15 @@ void window::signal_handler_sdl_key_down(const event::ui_event event,
 		}
 	}
 	if(!enter_disabled_ && (key == SDLK_KP_ENTER || key == SDLK_RETURN)) {
-		if (mod == 0) {
-			set_retval(retval::OK);
-			handled = true;
-		} else {
+		if (mod & (KMOD_CTRL | KMOD_ALT | KMOD_GUI | KMOD_SHIFT)) {
 			// Don't handle if modifier is pressed
 			handled = false;
+		} else {
+			// Only trigger if enter is not coming from multiline text
```

Comment:
This is pretty ugly… is it truly necessary? Without this, you can still insert a newline with Shift+Enter, can't you? The goal I was going for was for Shift+Enter to always work to insert a newline, but if you set `enter_disabled_` in the dialog then you can also use plain Enter.

For this retval behaviour I thought it should trigger on plain Enter only, but that doesn't mean I also expect Alt+Enter or Ctrl+Enter to insert a newline in a multiline text field.

Reply:
> This is pretty ugly… is it truly necessary? Without this, you can still insert a newline with Shift+Enter, can't you? The goal I was going for was for Shift+Enter to always work to insert a newline, but if you set `enter_disabled_` in the dialog then you can also use plain Enter.

Okay, I understand.
 
> For this retval behaviour I thought it should trigger on plain Enter only, but that doesn't mean I also expect Alt+Enter or Ctrl+Enter to insert a newline in a multiline text field.

So you want only Shift+Enter to insert a newline, but do not want Alt/Ctrl+Enter to insert newline, right?

Reply:
It's not terrible if it "just happens" to work with Alt and Ctrl because we didn't bother to avoid it, but I'd say the official shortcut should be Shift+Enter, or plain Enter if the dialog disables its Enter handling.

This also leaves open the possibility of making Ctrl+Enter or Alt+Enter do other things, if there's reason to do so in some dialog. I don't have any use-cases for that though.

Reply:
Implemented.